### ISSUE TITLE
Automatically benchmark for speed all encoders for inference and training

### ISSUE THREAD

**Issue by rom1504** [mlfoundations/open_clip#224]
related https://github.com/mlfoundations/open_clip/pull/97

**Comment by rom1504** [comment]
report on readme afterwards to estimate costs

**Comment by usuyama** [comment]
Love the idea!

Would be awesome to measure the memory requirements as well if possible, especially for training.

**Comment by rom1504** [comment]
yeah, would be good to have a script to do it automatically, then it could be possible to run it on some a100

**Comment by rom1504** [comment]
could also get memory usage

**Comment by usuyama** [comment]
Would be great to get the memory usage breakdown (A=model weights, B=forward pass), so we can estimate the memory usage of large batch size required for contrastive learning.
Estimate: A * 4 (weights, gradients, optim * 2) + B * batch_size.

I saw related threads in timm and huggingface:
https://github.com/rwightman/pytorch-image-models/issues/955
https://github.com/pytorch/torchdynamo/issues/425
